addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -2166,7 +2166,16 @@ export class OdooEditor extends EventTarget {
         // Determine colors at cursor position
         const sel = this.document.getSelection();
         if (sel.rangeCount && (!foreColor || !hiliteColor)) {
-            const endContainer = closestElement(sel.getRangeAt(0).endContainer);
+            // Reset the selection inside the editable
+            let range = sel.getRangeAt(0);
+            if (!this.editable.contains(range.endContainer)) {
+                range = new Range();
+                range.selectNodeContents(this.editable);
+                range.collapse(false);
+                sel.removeAllRanges();
+                sel.addRange(range);
+            }
+            const endContainer = closestElement(range.endContainer);
```

Comment:
@cammarosano well yes by doing this i will succeed to prevent traceback but the problem will still persist when i will click angle scroller the selection in [`_getSelectedColor`](https://github.com/odoo/odoo/blob/17.0/addons/web_editor/static/src/js/wysiwyg/wysiwyg.js#L1939)  will still be the same resulting in slider to disappear.Meaning i will still need to reset the selection back to editor somehow to solve this issue. Hence the fix i made here by resetting the selection back to editor which solved both our cases.